### DISCUSSION TITLE
Regarding Upgradation of Tensorflow

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2354] [Q&A] [answered]
Hi,

Is DeePMD going to follow Tensorflow 1 Computational graph approach only for training? Is there any specific reason for not incorporating Tensorflow 2 Keras Library?

**Comment by njzjz** [comment, marked as answer]
Graphs are still available in TF 2. It's faster than the eager execution. See https://www.tensorflow.org/guide/intro_to_graphs
